services/cron/tasks_extended.go

```diff
@@ -160,7 +160,9 @@ func initExtendedTasks() {
 	registerGarbageCollectRepositories()
 	registerRewriteAllPublicKeys()
 	registerRewriteAllPrincipalKeys()
-	registerRepositoryUpdateHook()
+	if !setting.DisableGitHooks {
+		registerRepositoryUpdateHook()
+	}
```

Comment:
I am not sure whether this logic is fine. Could there be a case like this:

1. User  sets up  the Git Hooks
2. The hooks are written into the repositories
3. User sets the DISABLE_GIT_HOOKS=true
4. **Then  all git  hooks are left in the repositories.**

Is this case expected?

Reply:
Logic is to disable all unnecessary GitHook items if GitHooks are disabled (i.e. UI elements and background jobs). Just added comment in case one need to clean up existing hooks before disabling GitHooks: 708319e4e9e2c9e797cc7dccfaa9ec7fe9a210eb